external/storm-hive/src/main/java/org/apache/storm/hive/bolt/HiveBolt.java

```diff
@@ -87,7 +87,7 @@ public void prepare(Map conf, TopologyContext topologyContext, OutputCollector c
                 }
             }
             this.collector = collector;
-            allWriters = new HashMap<HiveEndPoint,HiveWriter>();
+            allWriters = new ConcurrentHashMap<HiveEndPoint,HiveWriter>();
```

Comment:
We are running heartbeat as separate timer thread. Every time we add a new HiveEndPoint, HiveWriter we add this to allWriters Map and heartbeat thread goes through all the values to call heartbeat method. To have the values visible  in both threads we need to make concurrent